src/components/Footer/Footer.js

```diff
@@ -63,23 +87,133 @@ const Footer = ({
+      icon: (
+        <IcDmAnon
+          color={
+            !userAllowDm || loading.loadingGetAllowAnonDmStatus || loading.loadingDmAnon
+              ? COLORS.gray
+              : 'black'
+          }
+        />
+      ),
+      onPress: userAllowDm
+        ? onPressDMAnon
+        : () => {
+            Toast.show('User not allow to DM from Incognito Mode', Toast.SHORT);
+          },
+      style: (!userAllowDm || loading.loadingGetAllowAnonDmStatus || loading.loadingDmAnon) && {
+        color: COLORS.gray300
+      }
+    }
+  ];
+
   return (
     <BlurredLayer toastOnly={true} isVisible={item?.isBlurredPost}>
       <View style={[styles.rowSpaceBetween, styles.container]}>
         <View style={styles.leftGroupContainer}>
-          <TouchableOpacity
-            testID="shareBtn"
-            disabled={item?.isBlurredPost}
-            style={styles.btn}
-            onPress={onPressShare}>
-            <View style={styles.btnShare}>
-              <MemoIc_share
-                color={item?.isBlurredPost ? COLORS.gray : undefined}
-                height={20}
-                width={21}
-              />
-            </View>
-          </TouchableOpacity>
+          {isShowDM && !isSelf ? (
+            <TouchableOpacity
+              testID="sendDM"
+              style={[
+                {
+                  width: 100
+                }
+              ]}
+              onPress={async () => {
+                try {
+                  refSheet.current.open();
+                  setLoading({...loading, loadingGetAllowAnonDmStatus: true});
+                  const data = await getAllowAnonDmStatus(item.id);
+                  setUserAllowDm(data?.user.allow_anon_dm);
+                } catch (e) {
+                  console.log(e);
+                } finally {
+                  setLoading({...loading, loadingGetAllowAnonDmStatus: false});
+                }
+              }}>
+              <View
+                style={[
+                  styles.btnShare,
+                  {
+                    flexDirection: 'row',
+                    alignItems: 'center',
+                    paddingLeft: 0,
+                    paddingRight: 0
+                  }
+                ]}>
+                <MemoIc_senddm height={20} width={21} />
+                <Text
+                  style={[
+                    styles.text,
+                    {
+                      marginLeft: 4,
+                      fontWeight: '500'
+                    }
+                  ]}>
+                  {loading.loadingGetAllowAnonDmStatus ? 'Loading...' : 'DM'}
+                </Text>
+              </View>
+            </TouchableOpacity>
+          ) : (
+            <TouchableOpacity testID="shareBtn" style={styles.btn} onPress={onPressShare}>
+              <View style={styles.btnShare}>
+                <MemoIc_share height={20} width={21} />
+              </View>
+            </TouchableOpacity>
+          )}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There's a lot going on in this render method, which makes it hard to understand at a glance. Consider breaking down this logic into smaller, well-named helper functions or even custom hooks. This will make your code easier to read and maintain.



<!-- This is an auto-generated comment by OSS CodeRabbit -->